### PR TITLE
fix: account for negative timestamps from rrweb

### PR DESCRIPTION
**Pull request by daibhin**
We increasingly use the contents of rrweb events during ingestion so we need stricter validation...

We recently saw an issue caused by having received rrweb events from a client that had negative timestamps 🙃 

This change ensures that timestamps we lift from rrweb events to create replay events are positive and result in a valid luxon datetime

### CODE REVIEW

File: plugin-server/src/worker/ingestion/process-event.ts [hunk 1]
```diff
@@ -406,6 +406,7 @@ export const createSessionReplayEvent = (
 ) => {
     const timestamps = events
         .filter((e) => !!e?.timestamp)
+        .filter((e) => Math.sign(e.timestamp) === 1)
         .map((e) => castTimestampOrNow(DateTime.fromMillis(e.timestamp), TimestampFormat.ClickHouse))
```
Comment (pauldambra) [inline plugin-server/src/worker/ingestion/process-event.ts:410]:
```suggestion
        .filter((e) => Math.sign(e.timestamp) === 1)
        .map((e) => DateTime.fromMillis(e.timestamp))
        .filters((e) => e.isValid)
        .map((e) => castTimestampOrNow(e, TimestampFormat.ClickHouse))
```

(fly-by - I promise I'm not working 🙈)

we could add an explicit check?

https://github.com/moment/luxon/blob/master/docs/validity.md

Comment (pauldambra) [reply plugin-server/src/worker/ingestion/process-event.ts:410]: (core written in a GitHub comment, so definitely needs checking 🤣)

File: plugin-server/src/worker/ingestion/process-event.ts [hunk 2]
```diff
@@ -406,7 +406,10 @@ export const createSessionReplayEvent = (
 ) => {
     const timestamps = events
         .filter((e) => !!e?.timestamp)
-        .map((e) => castTimestampOrNow(DateTime.fromMillis(e.timestamp), TimestampFormat.ClickHouse))
+        .filter((e) => Math.sign(e.timestamp) === 1)
```
Comment (pauldambra) [inline plugin-server/src/worker/ingestion/process-event.ts:409]: I wonder if these two are unnecessary with the the validation below? Two fewer operations in a relatively hot path (which assumes `e.isValid` is cheap 🙈)

File: plugin-server/src/worker/ingestion/process-event.ts [hunk 3]
```diff
@@ -406,7 +406,10 @@ export const createSessionReplayEvent = (
...
+        .map((e) => DateTime.fromMillis(e.timestamp))
```
Comment (pauldambra) [inline plugin-server/src/worker/ingestion/process-event.ts:410]: So we'd just call `DateTime.fromMillis(e?.timestamp)` and then check validity

Arguably, then we should have tests on this specifically to simplify the tests but 🤷

**Review by pauldambra — APPROVED** [review]
Thoughts on how we could level up the validation even more but not blocking since this is better already

File: plugin-server/src/worker/ingestion/process-event.ts (hunk 2 above)
Comment (pauldambra) [reply plugin-server/src/worker/ingestion/process-event.ts]: They are not unnecessary!

**Comment by daibhin** [comment]
@pauldambra I didn't exactly know how to do it but another quick win would be to catch any invalid parameter and send it to Sentry. Debugging via logs / Kafka messages is always needle in a haystack kind of work

**Pull request merged**